arcaflow_plugin_fio/fio_schema.py

```diff
@@ -56,48 +65,90 @@ def is_sync(self) -> bool:
         return self.value in self._sync_io_engines
 
 
+time_pattern = re.compile(r"^\d*[d,h,m,s,ms,us]*$")
+
+size_pattern_string = (
+    r"(?:0x\d+)|"
+    r"(?:\d+(?:k|kb|ki|kib|m|mb|mi|mib|g|gb|gi|gib|t|tb|ti|tib|p|pb|pi|pib)*)"
+)
+size_pattern = re.compile(
+    rf"^{size_pattern_string}(?:,{size_pattern_string}){{0,2}}$", re.IGNORECASE
+)
+
+size_pattern_with_percent_string = (
+    rf"{size_pattern_string}|(?:[1-9][0-9]?%$|^100%)"
+)
+size_pattern_with_percent = re.compile(
+    rf"^{size_pattern_with_percent_string}$", re.IGNORECASE
+)
+
+size_range_pattern = re.compile(
+    rf"^(?:{size_pattern_string})-(?:{size_pattern_string})$",
+    re.IGNORECASE,
+)
```

Comment:
If you accept my suggestion for `size_pattern_string` above, I believe that you can dispense with the groupings here.